batch/batch/globals.py

```diff
@@ -1,2 +1,14 @@
 
+states = {'Pending', 'Ready', 'Running', 'Cancelled', 'Error', 'Failed', 'Success'}
+
 complete_states = ('Cancelled', 'Error', 'Failed', 'Success')
+
+valid_state_transitions = {
+    'Pending': {'Ready'},
+    'Ready': {'Running', 'Error', 'Cancelled'},
```

Comment:
These first two could jump to Success in rare cases that a second batch ran a pod to completion but died before reaping it. Then this batch learns of the pods existence.

Reply:
I ignore pods if I'm pending:

```
    if job and job._state == 'Pending':
        if pod:
            log.error('job {job.full_id} has pod {pod.metadata.name}, ignoring')
        return
```

If another batch started it, it means my data is stale, and I should just retry later.

I think I'm inclined to get rid of the distinct between Ready and Running.  Basically, if we have a pod in hand (or None), we know it is running (or ready), and this state seems redundant.  Will have the changes ready shortly.